utils/commands.js

```diff
@@ -0,0 +1,199 @@
+const rimraf = promisify(require('rimraf'));
+const boxen = require('boxen');
+const ora = require('ora');
+const process = require('process');
+
+const { readDotRemake, writeDotRemake, generateDotRemakeContent } = require('./dot-remake');
+const { registerUser, checkSubdomain, registerSubdomain, createDeploymentZip, removeDeploymentZip, pushZipToServer } = require('./helpers');
+const { questions } = require('./inquirer-questions');
+const { getSuccessMessage } = require('./get-success-message');
+
+const boxenOptions = {padding: 3, margin: 1, borderStyle: 'double', borderColor: 'green'};
+let spinner = null;
+
+const create = async (projectDir, options) => {
+  const cwd = process.cwd();
+  const newProjectDirPath = path.join(cwd, projectDir);
+  let rimrafError = null;
+
+  if (fs.existsSync(newProjectDirPath)) {
+    log(chalk.bgRed("Error: Cannot write to a directory that already exists."));
+    process.exit();
+  }
+
+  // STEP 1
+  spinner = ora("Creating new project.").start();
+  shell.exec(`git clone --depth 1 https://github.com/panphora/remake-framework.git ${projectDir}`, { silent: true });
+  spinner.succeed();
+
+  // STEP 2a & 2b
+  spinner = ora("Tidy up new project directory.").start();
+  rimrafError = await rimraf(path.join(newProjectDirPath, ".git"));
+
+  if (rimrafError) {
+    spinner.fail(chalk.bgRed("Error: Couldn't remove old .git directory from new project."));
+    process.exit();
+  }
+  spinner.succeed();
+
+  // put project README in the right place
+  shell.mv(path.join(newProjectDirPath, "README-FOR-BUNDLE.md"), path.join(newProjectDirPath, "README.md"));
+
+  // STEP 3
+  spinner = ora("Installing npm dependencies.").start();
+  shell.cd(newProjectDirPath);
+  shell.exec("npm install", { silent: true });
+  spinner.succeed();
+
+  // STEP 4
+  // write project name and env variables to .remake file
+  spinner = ora("Setting up .remake").start();
+  const dotRemakeObj = {
+    ...generateDotRemakeContent(options.multitenant)
+  }
+  spinner.succeed();
+
+  const dotRemakeReady = writeDotRemake(dotRemakeObj);
+
+  if (dotRemakeReady) {
+    log(boxen(getSuccessMessage(projectDir), boxenOptions));
+  }
+}
+
+const clean = () => {
+  let dotRemakeObj = readDotRemake();
+  if (!dotRemakeObj) {
+    log(chalk.bgRed('You are not in the root directory of a remake project.'));
+    process.exit();
+  }
+  spinner = ora('Cleaning project.').start();
+  shell.rm('-rf', '.cache/');
+  shell.rm('-rf', '_remake/dist');
+  spinner.succeed();
+  // TODO - replace above statement with bellow statement once the framework is 
+  // updated with the clean script
+  // shell.exec('npm run clean');
+}
+
+const build = () => {
+  let dotRemakeObj = readDotRemake();
+  if (!dotRemakeObj) {
+    log(chalk.bgRed('You are not in the root directory of a remake project.'));
+    process.exit();
+  }
+  spinner = ora('Building project.').start();
+  shell.exec('npm run build', { silent: true });
```

Comment:
What if the build fails? If it's silent, the user won't know of any errors.

Reply:
Good point! I will check if we can handle this in a better way.